### PR TITLE
use `AtomicBool` instead of `RwLock`

### PR DESCRIPTION
**Pull request by KnowWhoami**
This pr is a continuation of #274 which aims to use `AtomicBool` instead of `Rwlock`.

### CODE REVIEW

**Review by mojoX911 — APPROVED** [review]
Ack

File: tests/abort1.rs [hunk 1]
```diff
@@ -5,7 +5,7 @@ use coinswap::{
     taker::{SwapParams, TakerBehavior},
     utill::ConnectionType,
 };
-
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort1.rs:8]: Add this to std below

File: tests/abort1.rs [hunk 2]
```diff
@@ -195,14 +195,16 @@ fn test_stop_taker_after_setup() {
     taker_thread.join().unwrap();
 
     // Wait for Maker threads to conclude.
-    //makers.iter().for_each(|maker| maker.shutdown().unwrap());
+    // makers
+    //     .iter()
+    //     .for_each(|maker| maker.shutdown.store(true, Relaxed));
```
Comment (Shourya742) [inline tests/abort1.rs:200]: why do we have these comments,?

File: tests/abort2_case1.rs [hunk 3]
```diff
@@ -5,6 +5,7 @@ use coinswap::{
     taker::SwapParams,
     utill::ConnectionType,
 };
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort2_case1.rs:8]: move this

File: tests/abort3_case1.rs [hunk 4]
```diff
@@ -7,6 +7,7 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort3_case1.rs:10]: move

File: tests/abort3_case2.rs [hunk 5]
```diff
@@ -7,6 +7,7 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort3_case2.rs:10]: move

File: tests/abort3_case3.rs [hunk 6]
```diff
@@ -7,6 +7,7 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort3_case3.rs:10]: move

File: tests/fidelity.rs [hunk 7]
```diff
@@ -6,9 +6,10 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/fidelity.rs:9]: move

File: tests/malice1.rs [hunk 8]
```diff
@@ -7,6 +7,7 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/malice1.rs:10]: move

File: tests/malice2.rs [hunk 9]
```diff
@@ -7,6 +7,7 @@ use coinswap::{
 };
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/malice2.rs:10]: move

File: tests/standard_swap.rs [hunk 10]
```diff
@@ -10,6 +10,7 @@ use coinswap::{
 use bitcoind::bitcoincore_rpc::RpcApi;
 
 mod test_framework;
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/standard_swap.rs:13]: move

File: tests/abort2_case2.rs [hunk 11]
```diff
@@ -5,7 +5,7 @@ use coinswap::{
     taker::SwapParams,
     utill::ConnectionType,
 };
-
+use std::sync::atomic::Ordering::Relaxed;
```
Comment (Shourya742) [inline tests/abort2_case2.rs:8]: move

**Review by Shourya742 — CHANGES_REQUESTED** [review]
Can you test if there is any performance improvement or not?

**Review by Shourya742 — CHANGES_REQUESTED** [review]
I see you’ve marked all requested changes as resolved, but I don’t see any of them implemented. Could you add the changes and rebase? It looks like this PR has some conflicts, so please resolve those as well. It looks good to go once these changes are in place.

**Review by mojoX911 — APPROVED** [review]
reAck

**Comment by Shourya742** [comment]
> > Can you test if there is any performance improvement or not?
> 
> As we are handling small bunch of threads which are accessing these variables values at a fixed time interval so it has less probability of having collisions and since those mutex locks are hold and released in negligible time so I don't think so there will be much significant performance improvement.
> 
> Though this helps to simply our code structures by removing unnecessary complexity and api's.

Based on your this assertion, I would nack this PR.

**Review by Shourya742 — APPROVED** [review]

**Pull request merged**